### DISCUSSION TITLE
Can I get specific fields without graphql?

### DISCUSSION THREAD

**Discussion by enzzoperez** [strapi/strapi#7118] [Legacy Topics]
Hello, I need get only some fields for a relation, for example: I get this from an endpoint

```
[
  {
   name: 'bla',
   address: {
    id: 'ldjaslk',
    street: 'alsdjasjd',
    number: 90,
    cp: 4000,
    image: {...}
   }
 }
]
```
but I only need 

```
[
  {
   name: 'bla',
   address: {
    id: 'ldjaslk',
    street: 'alsdjasjd',
   }
 }
]
```

in the project we dont set up graphql, so I would like to know if it is possible using querys or something more to get that?

Thanks!

**Comment by sandiprb** [comment]
@enzzoperez You can have a look at this https://strapi.io/documentation/v3.x/concepts/services.html#core-services

example from the docs:

```
module.exports = {
  /**
   * Promise to fetch all records
   *
   * @return {Promise}
   */
  find(params, populate) {
    return strapi.query('restaurant').find(params, populate);
  },
};

```

**Look in the below part**

populate (array): you have to mention data you want populate ```["author", "author.name", "comment", "comment.content"]```

**Reply by enzzoperez** [reply]
Hello @sandiprb , I already found that I was search, https://strapi.io/documentation/3.0.0-beta.x/guides/custom-data-response.html#get-the-data-back .. Thank for response! :+1:

**Reply by sandiprb** [reply]
@enzzoperez glad to hear that. Please mark the discussion as answered :)
